### ISSUE TITLE
Possible memory leak

### ISSUE THREAD

**Issue by pdugan20** [katanyaJATI/react-native-image-blur-loading#4]
I keep getting the following warning with this component:

```
Warning: Can't perform a React state update on an unmounted component. This is
a no-op, but it indicates a memory leak in your application. To fix, cancel all
 subscriptions and asynchronous tasks in %s.%s, a useEffect cleanup function,
    in ImageBlurLoading (at Thumbnail.js:63)
```

**Comment by guilherme-ferraz1** [comment]
same

**Comment by RutujTailor28** [comment]
did you find any solution?

**Comment by farazaulia** [comment]
This error often happens when you still have pending asynchronous task and try update component's state but the component umounts or the app already render another component.

Here's some solution you can follow
[https://stackoverflow.com/questions/58038008/how-to-stop-memory-leak-in-useeffect-hook-react/58038029](https://stackoverflow.com/questions/58038008/how-to-stop-memory-leak-in-useeffect-hook-react/58038029)

**Comment by datoonw13** [comment]
cleanup is missing.
`  useEffect(() => {
    thumbnailSource && setHelper(true)
    const tu = setTimeout(() => setLoading(false), 20000) // fetch image timeout 20s
    return () => {
      clearTimeout(tu);
    }
  }, [])`

**Comment by FrederickEngelhardt** [comment]
```
  useEffect(() => {
    thumbnailSource && setHelper(true)
    const timer = setTimeout(() => setLoading(false), 20000) // fetch image timeout 20s
    return () => {
      if (timer){
        clearTimeout(timer)
      }
    }
  }, [])
```

**Comment by katanyaJATI** [comment]
Hi @pdugan20 @guilherme-ferraz1 @RutujTailor28 @datoonw13, sorry for the slow reply, and thanks for the feedback!
Also thanks to @FrederickEngelhardt & @hxffy for helping find a solution. 

The issue has been resolved in version 1.1.5.
Enjoy! 😉
